File: Scripts/server/ackermann-function/ruby/ackermann-function.rb
def ack(m, n)
  if m == 0
    n + 1
  elsif n == 0
    ack(m-1, 1)
  else
    ack(m-1, ack(m, n-1))
  end
end

(0..1000000).each do |i|
  (0..3).each do |m|
    (0..3).each do |n|
      ack(m,n)
    end
  end
end

File: Scripts/server/ackermann-function/javascript/ackermann-function.js
function ack(m, n) {
 return m === 0 ? n + 1 : ack(m - 1, n === 0  ? 1 : ack(m, n - 1));
}

var r = 0;

for ( var i = 0; i < 1000000; i++) {
    for ( var m = 0; m <= 3; m++) {
        for ( var n = 0; n <= 3; n++) {
            r = ack(m,n);
        }
    }  
}
